drivers/i2c/i2c_esp32.c

```diff
@@ -238,6 +239,27 @@ static int i2c_esp32_recover(const struct device *dev)
 	return 0;
 }
 
+static void IRAM_ATTR i2c_esp32_configure_timeout(const struct device *dev)
+{
+	const struct i2c_esp32_config *config = dev->config;
+
+	if (config->scl_timeout > 0) {
+		i2c_sclk_t sclk = i2c_get_clk_src(config->bitrate);
+		uint32_t clk_freq_mhz = I2C_LL_CLK_SRC_FREQ(sclk);
+		uint32_t timeout_cycles = MIN(I2C_TIME_OUT_REG_V,
+					      clk_freq_mhz / MHZ(1) * config->scl_timeout);
+		sys_clear_bits(I2C_TO_REG(config->index), I2C_TIME_OUT_REG);
+		sys_set_bits(I2C_TO_REG(config->index), timeout_cycles << I2C_TIME_OUT_REG_S);
```

Comment:
Did you want to mask or limit the maximum timeout here? Doesn't seem bound anywhere other than by the shift. If timeout_cycles is large and a shift left is done the timeout could unexpectedly be quite low no?

Reply:
It's already bound by the `MIN(a, b)` above, which limits the value to the maximum amount of bits allowed for that section of the register. Also the `I2C_TIME_OUT_REG_S` is actually 0 for all current ESP32 chips, but wanted to keep it generic.